frontend/src/metabase/common/hooks/use-search-list-query/use-search-list-query.ts

```diff
@@ -10,9 +10,12 @@ import type {
 } from "../use-entity-list-query";
 import { useEntityListQuery } from "../use-entity-list-query";
 
-export const useSearchListQuery = (
+export const useSearchListQuery = <
+  TItem = CollectionItem,
+  TResult = Omit<SearchResults, "data">,
+>(
   props: UseEntityListQueryProps<SearchListQuery> = {},
-): UseEntityListQueryResult<CollectionItem, Omit<SearchResults, "data">> => {
+): UseEntityListQueryResult<TItem, TResult> => {
   return useEntityListQuery(props, {
     fetchList: Search.actions.fetchList,
     getList: Search.selectors.getList,
```

Comment:
Per my [conversation on Slack](https://metaboat.slack.com/archives/C505ZNNH4/p1705603270151099) with @kulyk, I've modified `useSearchListQuery` so that it optionally takes a type. For example, if you do `const { data } = useSearchListQuery<SearchResult>()` then `data` will have the type `SearchResult[]`. If you do `const { data } = useSearchListQuery()` (a pattern found elsewhere in the codebase), `data` will have the type `CollectionItem[]` by default.